### ISSUE TITLE
Risk L2 Error - 'some complex event probabilities exceed 1'

### ISSUE THREAD

**Issue by blhumphreys** [NRCan/CanFlood#46]
Hi,

I am trying to run an L2 risk model using 7 hazard rasters and complex failure polygons. I am able to validate the model and run impacts L2, but when I run risk L2 I get the following error: 'some complex event probabilities exceed 1'. 

I have tried a different event_rels assumption (mutually exclusive instead of independent) and have confirmed that my exlikes values are all small (max 0.1659), as suggested by @cefect in issue #38.

Could you please help troubleshoot this issue?

Thanks.

![image](https://user-images.githubusercontent.com/100965696/167211105-07897087-582b-4503-b1ef-9497b5a44726.png)

**Comment by cefect** [comment]
You need to check all the values in a row of the exlikes (not individual values). For mutex, the row total needs to be less than 1. did you try with event_rels='max'?

Did you check the log file? there may be some more info there. 

Check these sections:
https://canflood.readthedocs.io/en/latest/applications_and_workflows.html#event-relations
https://canflood.readthedocs.io/en/latest/toolsets.html#conditional-p
https://canflood.readthedocs.io/en/latest/toolsets.html#risk-l2

**Comment by blhumphreys** [comment]
All of the rows in the exlikes files have a total of less than 1.

I ran with event_rels='max' (with mutually exclusive checked under Conditional P) and the resulting exlikes file also totals of less than 1. I did get the 'x and y arrays must have at least 2 entries' error though (all of my files have the same amount of hazard events).

Regarding the log file, it threw two warnings (for the mutex run):
- cf.ModelDialog.check_evals: received 1 unique events. this small number could cause unexpected behavior of the EAD algorhihtim
- cf.ModelDialog.set_exlikes: aep0.0050 failed 9449 (of 9449) Psum

Thanks for the user manual links.

**Comment by cefect** [comment]
Are you still getting errors? 

 I thought you had ~6 events (but the above suggests you're using 1). You need more than 1 event for any risk analysis. 

Be careful to distinguish between the event_rels of the [failure  polygon sampler ](https://canflood.readthedocs.io/en/latest/toolsets.html#conditional-p) and that specified in the control file for the [Risk L2 model](https://canflood.readthedocs.io/en/latest/toolsets.html#risk-l2)

**Comment by blhumphreys** [comment]
Yes, I am still getting errors. I am running 6 events (and have confirmed that this is reflected in the model build files). If helpful in diagnosing, the below screenshot shows the complete log file outputs for the MutEx Risk L2 run.

Thanks for linking to those sections of the manual. Good to know.

![image](https://user-images.githubusercontent.com/100965696/167280620-1b76f138-30bb-4524-8512-63d8feb5ab66.png)

**Comment by cefect** [comment]
OK, send me your model package again and I'll take another look. If you can only send the active files that would be helpful. but make sure everything needed to run the model is there.

**Comment by blhumphreys** [comment]
Sounds good, thanks @cefect. I have just emailed the model package over to you.

Please let me know if you need any more information.

**Comment by cefect** [comment]
files received. I see that you only have 1 hazard event (in 'evals) with 6 companion failure events. This is probably what's causing the 'x and y arrays' problem again... as you can't calculate risk with just one event. If you still want the expected value of your one event (given the 6 failures), which CanFlood isn't set up for, you'll probably need some workaround by adding dummy hazard events. 

I'm still looking into whats causing the Psum error.

**Comment by blhumphreys** [comment]
Thanks, I have added two additional hazard events and was able to get the model to run with event_rels='max'. I am still getting the Psum error (and one warning for each event in the log file) for Independent and MutEx though.

Sounds good re the Psum error.

**Comment by blhumphreys** [comment]
In addition to this, we will need to run 6 breach scenarios for 3 different events (total of 18 rasters) for our analysis. Is there a way to run all 18 rasters together? It seems that the tool is either limited to 8 (max inputs for the Conditional P tab), or until the total for each row in the exlikes row is greater than 1.

Please let me know if you'd like me to post this as a new issue.
